network_simulator/link.py

```diff
@@ -95,3 +164,118 @@ def __eq__(self, other):
 
     def __ne__(self, other):
         return not self.__eq__(other)
+
+    def get_link_cost(self):
+        """
+        Returns link cost. For now, just returns static propagation delay.
+        :return: float of link delay time in seconds
+        """
+
+        # TODO consider average wait time for last N packets
+        return self.static_delay_sec
+
+    def push(self, packet, global_clock_sec):
+        """
+        Pushes packet onto own LinkBuffer
+
+        :param Packet packet: Packet to push
+        :param global_clock_sec: store when packet pushed
+        :return: True if pushed, False if not
+        """
+        return self.link_buffer.push(packet, global_clock_sec)
+
+
+
+class LinkSendEvent(Event):
+    """
+    Event representing a packet arriving at the Link
+    """
+
+    def __init__(self, link):
+        """
+        :ivar link: link that must send its next packet
+        :ivar buff_elem: temporary variable containing LinkBufferElement to send
```

Comment:
Maybe just initialize the self.buff_elem to None in the constructor so the comments match the constructor code.